temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -584,6 +554,8 @@ private void handleResult(
               executionContext, activityTask, activityHandlerResult.getTaskFailed().getFailure());
 
       if (retryDecision.doNextAttempt()) {
+        // Release slot before scheduling the next attempt
+        slotSupplier.releaseSlot(SlotReleaseReason.willRetry(), executionContext.getPermit());
```

Comment:
Shouldn't you  pass the failure here like you do for normal activities?

Reply:
Normal activities don't actually set an error - it's more like when _processing_ failed rather than the task failed that the `error` variant is used. So I think this is consistent.